### ISSUE TITLE
Help me understand what needs to be done to fix it >.<

### ISSUE THREAD

**Issue by ensikon** [niftools/blender_nif_plugin#291]
![Screenshot_286](https://user-images.githubusercontent.com/50462223/57486850-aab3c300-72b7-11e9-8b14-4d5aa55e16b1.png)
I tried to export body model to nif :с

**Comment by ensikon** [comment]
i just don't know how to put shaders... i tried to use the guide, but it does not work >.<

**Comment by neomonkeus** [comment]
The shader type is not set in the property panel.

```python
        if b_obj.niftools_shader.bs_shadertype == 'None':
            raise nif_utils.NifError("Export version expected shader. "
                                     "No shader applied to mesh '%s', these cannot be exported to NIF."
                                     " Set shader before exporting." % b_obj)
```

**Comment by TagnumElite** [comment]
In the objects you are trying to export, you must set your shader type.
![Select BS Shader](https://user-images.githubusercontent.com/6497721/57612630-7b31de80-7575-11e9-81cc-82f43e1aa8ad.gif)

**Comment by neomonkeus** [comment]
I think there is an issue where importing skyrim shaders is also producing duplicate materials.

**Comment by Wolfstorm321** [comment]
> In the objects you are trying to export, you must set your shader type.
> ![Select BS Shader](https://user-images.githubusercontent.com/6497721/57612630-7b31de80-7575-11e9-81cc-82f43e1aa8ad.gif)


This "Niftools Shader" is nowhere to be found. None of the properties tab have it.

**Comment by neomonkeus** [comment]
> > In the objects you are trying to export, you must set your shader type.
> > ![Select BS Shader](https://user-images.githubusercontent.com/6497721/57612630-7b31de80-7575-11e9-81cc-82f43e1aa8ad.gif)
> 
> This "Niftools Shader" is nowhere to be found. None of the properties tab have it.

What version of the plugin are you using. 
I suspect you are using a more recent version (in fact I hope to release a new version this weekend), which has the newer texture nodes support. 

@Candoran2 Is this still an issue based on the way things work now.

**Comment by Candoran2** [comment]
> > > In the objects you are trying to export, you must set your shader type.
> > > ![Select BS Shader](https://user-images.githubusercontent.com/6497721/57612630-7b31de80-7575-11e9-81cc-82f43e1aa8ad.gif)
> > 
> > 
> > This "Niftools Shader" is nowhere to be found. None of the properties tab have it.
> 
> Thanks for nudging this issue.
> I suspect you are using a more recent version (in fact I hope to release a new version this weekend), which has the newer texture nodes support.
> What version of the plugin are you using?
> 
> @Candoran2 Is this still an issue based on the way things work now.

Currently, that Niftools Shader panel should be found under the material (if you have applied one to your mesh, if you don't have a material it will simply export without a shader property):

![image](https://user-images.githubusercontent.com/45334438/115887159-f2ef3000-a451-11eb-947b-994ab066a099.png)

Same thing if you're using 2.7x:

![image](https://user-images.githubusercontent.com/45334438/115888668-765d5100-a453-11eb-8007-d0148ccb8831.png)

(BTW, that must be an old gif, since there haven't been many changes to the 2.7 branch for a while). 

It's still an 'issue' in the sense that you have to set it, and it fails if you don't (at least for Skyrim meshes).

As far as "I think there is an issue where importing skyrim shaders is also producing duplicate materials." (Neomonkeus), goes, I haven't been able to replicate it, though I didn't look into it too hard. If you can give me the steps to replicate and what goes wrong, I'd be happy to look into it and try to fix it, if possible.

**Comment by neomonkeus** [comment]
It was just import a nif and it would cause a second material to be added to the object.

**Comment by Wolfstorm321** [comment]
> > > In the objects you are trying to export, you must set your shader type.
> > > ![Select BS Shader](https://user-images.githubusercontent.com/6497721/57612630-7b31de80-7575-11e9-81cc-82f43e1aa8ad.gif)
> > 
> > 
> > This "Niftools Shader" is nowhere to be found. None of the properties tab have it.
> 
> Thanks for nudging this issue.
> I suspect you are using a more recent version (in fact I hope to release a new version this weekend), which has the newer texture nodes support.
> What version of the plugin are you using?
> 
> @Candoran2 Is this still an issue based on the way things work now.

I'm using 0.0.5. I tried to install 0.0.1 but it gave me errors. Blender version is 2.92. 
Here is how the object panel looks like: 

![Clipboard36](https://user-images.githubusercontent.com/54403019/115970767-fef2f480-a51a-11eb-914d-584c96ba7f2b.jpg)

I cant find the nif shader tab, and always get this error when I try to export: 
Export version expected shader. No shader applied to mesh '<bpy_struct, Material("Material_30") at 0x0000000007769188, evaluated>', these cannot be exported to NIF.Set shader before exporting.

**Comment by neomonkeus** [comment]
I am not sure if 2.92 will be compatible with the addon. 
Can you attach the files and steps you used to reproduce.

**Comment by Wolfstorm321** [comment]
> I am not sure if 2.92 will be compatible with the addon.
> Can you attach the files and steps you used to reproduce.

I found it, it is in another tab, the "material properties" tab: 

![Clipboard37](https://user-images.githubusercontent.com/54403019/115970878-c4d62280-a51b-11eb-84ce-070e349ad4ad.jpg)

Tutorials are pointing to the object tab, though. I was using a tutorial for 2.7.x.

**Comment by neomonkeus** [comment]
Yes, a 2.7 tutorial will not be useful for 2.8 series of blender.

Do you have 2 materials or not?

**Comment by Wolfstorm321** [comment]
Only one tab for materials, and some nif tabs have moved there in 2.9.

**Comment by neomonkeus** [comment]
Can you share the file.

**Comment by Wolfstorm321** [comment]
https://mega.nz/file/BAkUXQbK#DucHO-54FumRYwdFzICRaD9DlKKh1gt97WyZF4jO7sU
